src/main/ts/components/ChartVerticalLabel.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Clean as You Code Plugin
+ * Copyright (C) 2022-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import React from 'react';
+import { GRAPH_VERTICAL_LABEL_LEFT_MARGIN, GRAPH_VERTICAL_LABEL_Y_BASE_OFFSET } from '../constants';
+import Line from './Line';
+import VerticalLine from './VerticalLine';
+
+interface ChartVerticalLabelProps {
+  label: string;
+  level: number;
+  x: number;
+}
+
+export default function ChartVerticalLabel(props: ChartVerticalLabelProps) {
+  const { label, level, x } = props;
+  const textY = -GRAPH_VERTICAL_LABEL_Y_BASE_OFFSET * level;
```

Comment:
NP: So that space between label is not a function of the initial offset (= space with the graph). Of course it will need a proper constant declaration and maybe a rework of the level mechanism (starts at 0 instead of 1).

```suggestion
  const textY = -GRAPH_VERTICAL_LABEL_Y_BASE_OFFSET - (level - 1) * 20;
```

Reply:
Or we could say that `level = 0 = no offset` and  rename `level` to `labelYOffsetLevel`, WDYT?

Reply:
Sounds good